### DISCUSSION TITLE
Product page upload file(s) or add text for personalisation

### DISCUSSION THREAD

**Discussion by vasugarg09** [bagisto/bagisto#9756] [Q&A]
I want to create a store where the customer will be able to upload files and write text compulsorily before adding the product to the cart.
The number of files that need to be added compulsorily will depend on the variant that the customer chooses.
These 2 Bagisto websites are similar to what I want to achieve (but in my case the no of attachments will be dependent on the variant)
How can I achieve this?

**References:** 

* https://www.shopinireland.ie/personalised-21st-birthday-frame-gift
* https://prezentzgrawerem.com/naszyjnik-z-grawerem-prezent-dla-babci-latarnia---niesmiertelnik

**Example**

<img width="282" alt="Screenshot 2024-05-04 at 12 18 22 PM" src="https://github.com/bagisto/bagisto/assets/66393701/9cff5041-e93d-4239-ad6d-9111d3653736">

**Comment by devansh-webkul** [comment]
You can use the customizable option feature in the Bagisto latest version.
